### DISCUSSION TITLE
API/UI to delete dataset items and runs

### DISCUSSION THREAD

**Discussion by tomek82** [langfuse/langfuse#1987] [Ideas]
### Describe the feature or potential improvement

Sometimes a dataset item is added erroneously or a dataset run is corrupted because of wrong evaluations, configuration etc.

It would be nice if there was an option to manually clean up those things to limit the noise.

**Comment by jesusfbes** [comment]
I think that this is very interesting. Do anybody know if there is an API way to do it?

**Reply by ValentinKovalev** [reply]
I believe it's crucial to prioritize this feature. Currently, there is no option to delete dataset items and runs via the API or UI, which can be vital for debugging. I strongly support adding this capability.

**Comment by serlina** [comment]
yes, I also met this problem, I added the duplicated data items in the dataset, and I wanted to delete them, But I didn't find any ways to do this. now I set these items as archieved, and they will not be eveulated anymore, But I really want to delete such data from my dataset.

Before the delete function is provided, can we delete the dataset items from postgres database? I'm not sure this can be a workground, if yes, does anyone help to provide detail steps to do this in the database.

**Reply by marcklingen** [reply]
You can archive dataset items by updating the `status` to `ARCHIVED`

https://api.reference.langfuse.com/#post-/api/public/dataset-items
![CleanShot 2024-09-10 at 15 18 26](https://github.com/user-attachments/assets/4e29743b-5b10-4048-b962-6503f95018eb)

**Reply by ghost** [reply]
Dataset item archival in the UI page for an individual db item would be appreciated as well 
E.g. by adding an `Archive` button next to the `Save Changes` one.
![image](https://github.com/user-attachments/assets/c8623e37-8f48-4462-b4aa-55f6fca5e870)

**Reply by marliessophie** [reply]
@FlorDonnaSanders you can already archive single dataset items via the Actions column (shows three dots) on the dataset items table

**Comment by marcklingen** [comment]
All of these items can now be deleted via the API and UI, thanks again for raising this!

**Reply by Bukunmi2108** [reply]
kindly add for the sdk too. Sometimes, i want to able to do it programmatically without going to the ui any time there is a dataset change. And, I dont want to delete the entire dataset because that would clear the run history.

**Reply by ksafonov** [reply]
+1

**Comment by dwillems-nrb** [comment]
Hello,

As I see it is possible te delete a item from a dataset, ok.
But is it possible only to delete a dataset run item from a run, for example when 1 call failed (due to network error or something like that) in a looong dataqset evaluation ?
As I can see, I can GET or POST dataset-run-items but not delete it :(
